src/main/java/org/broadinstitute/hellbender/tools/examples/ExampleReadWalkerWithVariantsSpark.java

```diff
@@ -0,0 +1,89 @@
+package org.broadinstitute.hellbender.tools.examples;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.ExampleProgramGroup;
+import org.broadinstitute.hellbender.engine.FeatureContext;
+import org.broadinstitute.hellbender.engine.FeatureInput;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.spark.ReadWalkerContext;
+import org.broadinstitute.hellbender.engine.spark.ReadWalkerSpark;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.util.List;
+
+/**
+ * Example/toy program that prints reads from the provided file or files along with overlapping variants
+ * (if a source of variants is provided). Intended to show how to implement the ReadWalker interface.
+ */
+@CommandLineProgramProperties(
+        summary = "Prints reads from the provided file(s) along with overlapping variants (if a source of variants is provided) to the specified output file (or STDOUT if none specified)",
```

Comment:
You should add a comment to the read walker javadocs pointing to this class as an example